File: NYTPhotoViewerTests/NYTPhotoViewControllerTests.m
//
//  NYTPhotoViewControllerTests.m
//  NYTPhotoViewer
//
//  Created by Brian Capps on 2/24/15.
//  Copyright (c) 2015 NYTimes. All rights reserved.
//

@import UIKit;
@import XCTest;

#import <NYTPhotoViewer/NYTPhotoViewController.h>
#import "NYTExamplePhoto.h"

@interface NYTPhotoViewControllerTests : XCTestCase

@end

@implementation NYTPhotoViewControllerTests

- (void)testScalingImageViewExistsAferInitialization {
    NYTPhotoViewController *photoViewController = [[NYTPhotoViewController alloc] initWithPhoto:[self newTestPhoto] itemIndex:0 loadingView:nil notificationCenter:nil];
    XCTAssertNotNil(photoViewController.scalingImageView);
}

- (void)testDoubleTapGestureRecognizerExistsAferInitialization {
    NYTPhotoViewController *photoViewController = [[NYTPhotoViewController alloc] initWithPhoto:[self newTestPhoto] itemIndex:0 loadingView:nil notificationCenter:nil];
    XCTAssertNotNil(photoViewController.doubleTapGestureRecognizer);
}

- (void)testLoadingViewExistsAferNilInitialization {
    NYTPhotoViewController *photoViewController = [[NYTPhotoViewController alloc] initWithPhoto:[self newTestPhoto] itemIndex:0 loadingView:nil notificationCenter:nil];
    XCTAssertNotNil(photoViewController.loadingView);
}

- (void)testDesignatedInitializerAcceptsNilForPhotoArgument {
    XCTAssertNoThrow([[NYTPhotoViewController alloc] initWithPhoto:nil itemIndex:0 loadingView:[[UIView alloc] init] notificationCenter:[NSNotificationCenter defaultCenter]]);
}

- (void)testDesignatedInitializerAcceptsNilForLoadingViewArgument {
    XCTAssertNoThrow([[NYTPhotoViewController alloc] initWithPhoto:[self newTestPhoto] itemIndex:0 loadingView:nil notificationCenter:[NSNotificationCenter defaultCenter]]);
}

- (void)testDesignatedInitializerAcceptsNilForNotificationCenterArgument {
    XCTAssertNoThrow([[NYTPhotoViewController alloc] initWithPhoto:[self newTestPhoto] itemIndex:0 loadingView:[[UIView alloc] init] notificationCenter:nil]);
}

- (void)testDesignatedInitializerAcceptsNilForAllArguments {
    XCTAssertNoThrow([[NYTPhotoViewController alloc] initWithPhoto:nil itemIndex:0 loadingView:nil notificationCenter:nil]);
}

- (NYTExamplePhoto *)newTestPhoto {
    NYTExamplePhoto *testPhoto = [[NYTExamplePhoto alloc] init];
    return testPhoto;
}

@end

File: Example/NYTPhotoViewer-SwiftTests/NYTPhotoViewControllerTests.swift
//
//  NYTPhotoViewControllerTests.swift
//  NYTPhotoViewer
//
//  Created by Mark Keefe on 3/21/15.
//  Copyright (c) 2015 The New York Times. All rights reserved.
//

import XCTest

class NYTPhotoViewControllerTests: XCTestCase {
    
    var testPhoto: ExamplePhoto?

    override func setUp() {
        super.setUp()
        testPhoto = ExamplePhoto(attributedCaptionTitle: NSAttributedString())
    }
    
    override func tearDown() {
        testPhoto = nil
        super.tearDown()
    }

    func testScalingImageViewExistsAferInitialization() {
        let photoViewController = NYTPhotoViewController(photo: testPhoto, loadingView: nil, notificationCenter: nil)
        XCTAssertNotNil(photoViewController.scalingImageView)
    }
    
    func testDoubleTapGestureRecognizerExistsAferInitialization() {
        let photoViewController = NYTPhotoViewController(photo: testPhoto, loadingView: nil, notificationCenter: nil)
        XCTAssertNotNil(photoViewController.doubleTapGestureRecognizer)
    }
    
    func testLoadingViewExistsAferNilInitialization() {
        let photoViewController = NYTPhotoViewController(photo: testPhoto, loadingView: nil, notificationCenter: nil)
        XCTAssertNotNil(photoViewController.loadingView)
    }
    
    func testDesignatedInitializerAcceptsNilForPhotoArgument() {
        let photoViewController = NYTPhotoViewController(photo: nil, loadingView: UIView(), notificationCenter: NSNotificationCenter.defaultCenter())

        XCTAssertNotNil(photoViewController)
    }
    
    func testDesignatedInitializerAcceptsNilForLoadingViewArgument() {
        let photoViewController = NYTPhotoViewController(photo: testPhoto, loadingView: nil, notificationCenter: NSNotificationCenter.defaultCenter())
        
        XCTAssertNotNil(photoViewController)
    }
    
    func testDesignatedInitializerAcceptsNilForNotificationCenterArgument() {
        let photoViewController = NYTPhotoViewController(photo: testPhoto, loadingView: UIView(), notificationCenter: nil)
        
        XCTAssertNotNil(photoViewController)
    }
    
    func testDesignatedInitializerAcceptsNilForAllArguments() {
        let photoViewController = NYTPhotoViewController(photo: nil, loadingView: nil, notificationCenter: nil)
        
        XCTAssertNotNil(photoViewController)
    }
}
